dynamic_instrumentation/src/instrument_memory.rs

```diff
@@ -664,29 +664,31 @@ fn instrument<'tcx>(
         index: CRATE_DEF_INDEX,
     };
 
-    let mut instrumenter = FunctionInstrumenter {
+    let mut collect_points = CollectFunctionInstrumentationPoints {
         tcx,
         body,
         runtime_crate_did,
 
         instrumentation_points: RefCell::new(vec![]),
         rvalue_dest: None,
     };
-    let main_did = tcx.entry_fn(()).map(|(def_id, _)| def_id);
-    instrumenter.visit_body(body);
-    do_instrumentation(
+    collect_points.visit_body(body);
+    apply_instrumentation(
         state,
-        &instrumenter.into_instrumentation_points(),
+        &collect_points.into_instrumentation_points(),
         tcx,
         body,
         body_def_hash,
     );
 
+    // Apply `main`-specific instrumentation if this fn is main
+    let main_did = tcx.entry_fn(()).map(|(def_id, _)| def_id);
```

Comment:
This is a bit separate from the changes here, but `did` as an abbreviation for `def_id` is pretty bad.  `did` is another word, and `def_id` is very long, and both `def`, `id`, and `def_id` would make more sense than `did` to me.

Reply:
Agreed, but `did` for `DefId` is a naming convention already present in MIR.

Reply:
That's unfortunate, but it makes sense if they already do it.